### PR TITLE
sg.config.yaml: Add command enterprise-single-gitserver

### PR DESCRIPTION
**Pull request by indradhanush**
For times when we explicitly want a single gitserver instance to ease debugging a very specific issue.

## Test plan

New command set for `sg start`. Tested that command works locally.

### CODE REVIEW

File: sg.config.yaml [hunk 1]
```diff
@@ -809,6 +809,30 @@ commandsets:
...
+      - docker
+      - redis
+      - postgres
+      - git
+    commands:
+      - frontend
+      - worker
+      - repo-updater
+      - web
+      - gitserver-0
```
Comment (mrnugget) [inline sg.config.yaml:824]: That's not enough. You also need to make sure that these are changed:

https://github.com/sourcegraph/sourcegraph/blob/b7202cc4ac964234af0dcf349f060bda75db4f5c/sg.config.yaml#L20-L23

**Review by mrnugget — APPROVED** [review]
Out of curiosity: what's the usecase that's better debugged with a single gitserver instance?

**Comment by indradhanush** [comment]
> Out of curiosity: what's the usecase that's better debugged with a single gitserver instance?

Repo corruption with `sgm`. More like reproduction attempt instead of debugging so far.

**Comment by indradhanush** [comment]
To clarify, the issue is not dependent on one vs two gitserver instances, but more like with one gitserver running, it's easier to focus down on only that instance.

**Comment by indradhanush** [comment]
@mrnugget If you think this command could have negative effects of less devs using multiple shards locally, I'm happy to not merge it in.

**Comment by mrnugget** [comment]
> @mrnugget If you think this command could have negative effects of less devs using multiple shards locally, I'm happy to not merge it in.

I don't think there's negative effects, to be honest. But I also don't think there's a really strong reason to merge it. You can always add this kind of stuff to your local `sg.config.overwrite.yaml`, which is `.gitignore`d and allows you to setup stuff exactly like that.

**Comment by indradhanush** [comment]
Hmm. Didn't realize that I could do that for commandsets as well. I agree, no point in merging this if we can do this in the overwrite file. 👍

**Pull request closed without merging**